api/v4/source/content_flagging.yaml

```diff
@@ -0,0 +1,63 @@
+  /api/v4/content_flagging/flag/config:
+    get:
+      summary: Get content flagging configuration
+      description: |
+        Returns the configuration for content flagging, including the list of available reasons for flagging content. This data is used to gather details from the user when they flag content.
+        An enterprise advanced license is required.
+      tags:
+        - Content Flagging
+      responses:
+        '200':
+          description: Configuration retrieved successfully
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  reasons:
+                    type: array
+                    items:
+                      type: string
+                    description: List of reasons for flagging content
+                  reporter_comment_required:
+                    type: boolean
+                    description: Indicates if a comment from the reporter is required when flagging content
+        '500':
+          description: Internal server error.
+        '501':
+          description: Feature is disabled either via config or an Enterprise Advanced license is not available.
+        '404':
+          description: Feature is disabled via the feature flag.
+  /api/v4/content_flagging/team/{team_id}/status:
+    get:
+      summary: Get content flagging status for a team
+      description: |
+        Returns the content flagging status for a specific team, indicating whether content flagging is enabled on the specified team or not.
+      tags:
+        - Content Flagging
+      parameters:
+        - in: path
+          name: team_id
+          required: true
+          schema:
+            type: string
+          description: The ID of the team to retrieve the content flagging status for
+      responses:
+        '200':
+          description: Content flagging status retrieved successfully
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  enabled:
+                    type: boolean
+                    description: Indicates if content flagging is enabled for the team
+        '404':
+          description: The specified team was not found or the feature is disabled via the feature flag.
+        '500':
+          description: Internal server error.
+        '403':
+          description: Forbidden - User does not have permission to access this team.
+        '501':
+          description: Feature is disabled either via config or an Enterprise Advanced license is not available.
```

Comment:
nit: the order can be improved

Reply:
Done, ordered in ascending order of error code.